File: yukicoder/yuki246.py
left, right = 1, 10**9+1
while right - left > 1:
    mid = (left + right) // 2
    print('? {}'.format(mid), flush=True)
    if int(input()):
        left = mid
    else:
        right = mid
print('! {}'.format(left), flush=True)

File: yukicoder/yuki246.nim
from strutils import parseInt
var
  left = 1
  right = 1e9.toint + 1
while left + 1 < right:
  let mid = (left + right) div 2
  echo("? ", mid)
  flushFile(stdout)
  if stdin.readline.parseInt == 0:
    right = mid
  else:
    left = mid
echo("! ", left)
flushFile(stdout)
